analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -111,18 +104,23 @@ class NuGetSupport(serviceIndexUrls: List<String> = listOf(DEFAULT_SERVICE_INDEX
         }
     }
 
-    private val serviceIndices = runBlocking {
-        serviceIndexUrls.map {
-            async { JSON_MAPPER.readValueFromUrl<ServiceIndex>(it) }
-        }.awaitAll()
-    }
+    private val registrationsBaseUrls: List<String> = run {
+        val configFile = definitionFile.parentFile.searchUpwardsForFile("nuget.config", ignoreCase = true)
+        val serviceIndexUrls = configFile?.let { NuGetConfigFileReader.getRegistrationsBaseUrls(it) }
+            ?: listOf(DEFAULT_SERVICE_INDEX_URL)
+        val serviceIndices = runBlocking {
+            serviceIndexUrls.map {
+                async { JSON_MAPPER.readValueFromUrl<ServiceIndex>(it) }
+            }.awaitAll()
+        }
 
-    // Note: Remove a trailing slash as one is always added later to separate from the path, and a double-slash would
-    // break the URL!
-    private val registrationsBaseUrls = serviceIndices
-        .flatMap { it.resources }
-        .filter { it.type == REGISTRATIONS_BASE_URL_TYPE }
-        .map { it.id.removeSuffix("/") }
+        // Note: Remove a trailing slash as one is always added later to separate from the path, and a double-slash
+        // would break the URL!
```

Comment:
Nit: I know the comment has been at this place before, but now that you're moving this code anyway, how about also moving this comment just before the belonging line 122?

Reply:
Done in a commit on top.